src/template.c

```diff
@@ -153,11 +175,20 @@ std::vector<std::string> template_dfa(CharacterVector x) {
   }
 
   if (!(state == html || state == html_oneOpenBracket)) {
-    stop("HTML template did not end in html state (missing closing \"}}\").");
+    Rf_error("HTML template did not end in html state (missing closing \"}}\").");
   }
 
   // Add ending HTML piece
-  pieces.push_back(input.substr(pieceStartIdx, len - pieceStartIdx));
+  str = PROTECT(Rf_mkCharLenCE(input + pieceStartIdx, len - pieceStartIdx, CE_UTF8));
+  REPROTECT(pieces = str_sxp_set(pieces, pieces_num++, str), pieces_idx);
+  UNPROTECT(1);
+
+  if (pieces_num < Rf_xlength(pieces)) {
+    SETLENGTH(pieces, pieces_num);
+    SET_TRUELENGTH(pieces, pieces_num);
```

Comment:
These lines also seem weird, but setting the length like this does work and is the most efficient way to get a properly sized vector without an additional copy. This is what we do in [readr](https://github.com/tidyverse/readr/blob/d07eb692c412927fba47d319d8b66d9e64a82615/src/Collector.h#L48-L49)

Reply:
Can you add a comment about that?

Reply:
Done!